common/Util.cpp

```diff
@@ -125,15 +127,14 @@ namespace Util
 
             // a poor fallback but something.
             std::vector<char> random = getBytes(length);
-            int fd = open("/dev/urandom", O_RDONLY);
+            int fd = getrandom(random.data(), length, GRND_NONBLOCK);
```

Comment:
Is this proper usage of `getrandom()`? I made a lot of research on the internet but couldn't find any proper migrations from `/dev/[u]random` to `getrandom()/getentropy()` on other projects (in order to have an idea).

Reply:
The function is straightforward. I assume you spent some time reading the documentation at https://man7.org/linux/man-pages/man2/getrandom.2.html. If not, please read it a couple of times.

The return value is the number of bytes that was written. This is equivalent to `len` below. Furthermore, since `fd` was the file descriptor, which you have deleted, we also need to delete the `read` below, since there is no longer a file to read.